### PR TITLE
Add tests for ETL functionality

### PR DESCRIPTION
**Pull request by Nicoretti**

### CODE REVIEW

File: test/integration/etl_test.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    ddl = f"DROP TABLE IF EXISTS {name};"
+    connection.execute(ddl)
+    connection.commit()
+
+
+@pytest.mark.etl
+@pytest.mark.parametrize(
+    "params,expected",
+    [
```
Comment (tkilias) [inline test/integration/etl_test.py:53]: The parameters are partially hard to read, because they are so diverse

Comment (tkilias) [reply test/integration/etl_test.py:53]: And do you need expected? It seems always the same

File: test/integration/etl_test.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    query = cleandoc(
+        f"""
+        SELECT * FROM {table}
+        WHERE USER_ID NOT IN (SELECT USER_ID from {import_table});
+        """
+    )
+    result = connection.execute(query)
+
+    expected = []
```
Comment (tkilias) [inline test/integration/etl_test.py:87]: Is it really a good idea to actually do not import and export any data

File: test/integration/export_test.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+def swaped_data(data):
+    yield [(lastname, firstname) for firstname, lastname in data]
+
+
+@pytest.fixture
+def table(connection, empty_table, data):
+    name = empty_table
+    stmt = "INSERT INTO {table} VALUES ({{col1}}, {{col2}});"
+    for col1, col2 in data:
+        connection.execute(stmt.format(table=name), {"col1": col1, "col2": col2})
```
Comment (tkilias) [inline test/integration/export_test.py:55]: The format would be usually not the best idea, but can be ignored here.

File: test/integration/export_test.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+
+@pytest.fixture
+def expected_csv(csv_dialect):
+    def create_csv(table, data, **params):
+        csvfile = io.StringIO()
+        if "with_column_names" in params and params["with_column_names"]:
+            data = [("FIRSTNAME", "LASTNAME")] + data
+        writer = csv.writer(csvfile, csv_dialect)
+        writer.writerows(data)
```
Comment (tkilias) [inline test/integration/export_test.py:83]: Does the csv.writer needs to be closed?

File: test/integration/export_test.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    connection.commit()
+
+
+@pytest.fixture
+def data(faker):
+    yield [(faker.first_name(), faker.last_name()) for _ in range(0, 10)]
+
+
+@pytest.fixture
+def swaped_data(data):
```
Comment (tkilias) [inline test/integration/export_test.py:46]: I am not seeing this fixture used. Am I blind?

File: test/integration/import_test.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    file = tmp_path / "names.csv"
+    with open(file, "w", newline="") as csvfile:
+        writer = csv.writer(csvfile, dialect="unix")
+        for row in data:
+            writer.writerow(row)
+    yield file
+
+
+@pytest.mark.etl
+def test_import_with_custom_csv_format(connection, empty_table, csv_file, data):
```
Comment (tkilias) [inline test/integration/import_test.py:60]: How is this a custom csv format, it seems to unix dialect , is this something special?

**Review by tkilias — CHANGES_REQUESTED** [review]

File: test/integration/export_test.py (hunk 4 above)
Comment (Nicoretti) [reply test/integration/export_test.py:78]: AFIK no, it takes a file like object as it's first parameter. From my understanding the one providing the file like object is responsible to close the provided file-like object afterwards (later).

File: test/integration/import_test.py (hunk 6 above)
Comment (Nicoretti) [reply test/integration/import_test.py]: No maybe the name is not ideal, I don't recall why I'll rename it.

File: test/integration/etl_test.py (hunk 2 above)
Comment (Nicoretti) [reply test/integration/etl_test.py:84]: The expected parameter should be removed. In fact, a full export and import is conducted. 

Here is the concept of the test:

1. Export: `USERS` Table to `CSV File`
2. Import: `CSV File` to `USERS_IMPORT` Table
3. Verify that the sets/tables are identical

Comment (tkilias) [reply test/integration/etl_test.py:84]: a table is the table with data and input_table the empty target table

**Review by tkilias — APPROVED** [review]

**Pull request merged**